samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerIds.java

```diff
@@ -60,16 +54,19 @@ public GroupByContainerIds(int count) {
   }
 
   public Set<ContainerModel> group(Set<TaskModel> tasks, List<String> containersIds) {
+    if (containersIds == null)
+      return this.group(tasks);
+
+    if (containersIds.isEmpty())
+      throw new IllegalArgumentException("Must have at least one container");
+
```

Comment:
Not sure if its just me who feels slightly weird about this contract about mismatch is behavior.
Although I agree semantically `null` and `empty` lists mean two different things, I have often seen them being used interchangeably or at the least have similar behavior in many codebases. 
Just putting in a comment to see others take on it.

Feel free to drop my comment if rest of the team is fine with it.